### PR TITLE
Add support for x-terraform-field-name extension

### PR DESCRIPTION
**Pull request by dikhan**
## Proposed changes

Please check out issue description for more details about the feature request.

## Type of change

What type of change does your code introduce to the provider? Please put an `x` (w/o heading/trailing white spaces) 
in the boxes that apply:

- [ ] Bugfix (change that fixes current functionality)
- [x] New feature (change that adds new functionality)

Does this Pull Request resolve any open issue? If so, please make sure to link to that issue:

Fixes: #41 

## Checklist

Please put an `x` (w/o heading/trailing white spaces) in the boxes that apply:

- [x] I have read and followed the [CONTRIBUTING](https://github.com/dikhan/terraform-provider-api/blob/master/.github/CONTRIBUTING.md) guidelines
- [x] I have made sure code compiles correctly
- [x] I have run 'make test' locally from the terraform_provider_api folder and no errors were found
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added/updated necessary documentation (if appropriate)

### CODE REVIEW

File: docs/how_to.md [hunk 1]
```diff
@@ -394,6 +394,7 @@ x-terraform-immutable | boolean |  The field will be used to create a brand new
...
 x-terraform-sensitive | boolean |  If this meta attribute is present in an object definition property, it will be considered sensitive as far as terraform is concerned, meaning that its value will not be disclosed in the TF state file
 x-terraform-id | boolean | If this meta attribute is present in an object definition property, the value will be used as the resource identifier when performing the read, update and delete API operations. The value will also be stored in the ID field of the local state file.
+x-terraform-field-name | string | This enables service providers to override the schema definition property name with a different one which will be the property name used in the terraform configuration file. This is mostly used to expose the internal property to a more user friendly name. If the extension is not present and the property name is not terraform compliant, an automatic convertion will be performed by the OpenAPI Terraform provider to make the name compliant (following Terraform's field name convention to be snake_case)
```
Comment (pschuegr) [inline docs/how_to.md]: convertion => conversion

File: examples/swaggercodegen/api/api/auth.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"net/http"
+	"fmt"
+	"errors"
+)
+
+func AuthenticateRequest(r *http.Request, w http.ResponseWriter) error {
+	apiKey := r.Header.Get("Authorization")
+	if apiKey == "" || apiKey != "apiKeyValue" {
```
Comment (pschuegr) [inline examples/swaggercodegen/api/api/auth.go]: Is the first part of this necessary?  if it's an empty string, it's not going to be `"apiKeyValue"`

Comment (pschuegr) [inline examples/swaggercodegen/api/api/auth.go]: Is the . first part of this necessary? if string is empty it will also fail the second test.

Comment (dikhan) [reply examples/swaggercodegen/api/api/auth.go]: agreed, getting rid of that redundant code right away :)

**Pull request merged**